src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -200,6 +204,52 @@ private TypeSymbol DecodeType(TypeSymbol type)
             }
         }
 
+        private PointerTypeSymbol DecodePointerType(PointerTypeSymbol type)
+        {
+            return type.WithPointedAtType(DecodeTypeInternal(type.PointedAtTypeWithAnnotations));
+        }
+
+        private FunctionPointerTypeSymbol DecodeFunctionPointerType(FunctionPointerTypeSymbol type)
```

Comment:
>DecodeFunctionPointerType [](start = 42, length = 25)

We don't need to make any changes for proper encoding?

Reply:
Thankfully, no. The tuple names encoding is very straightforward and does not need to look at custom modifiers or have a special ordering, and a standard VisitTypes implementation (already done) encoded correctly.

---
In reply to: [420411876](https://github.com/dotnet/roslyn/pull/43961#discussion_r420411876) [](ancestors = 420411876)